### PR TITLE
add Uploader component

### PR DESCRIPTION
**Pull request by mouseluka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+<p align="center">
```
Comment (igor-lemon) [inline README.md:1]: Car Readme. Add a simple example how to use the component

File: components/upload/Upload.tsx [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+  React.useEffect(() => {
+    devWarning(
+      'fileList' in props || !('value' in props),
+      'Upload',
+      '`value` is not a valid prop, do you mean `fileList`?',
+    );
+  }, []);
+
+  const onChange = (info: UploadChangeParam) => {
```
Comment (igor-lemon) [inline components/upload/Upload.tsx:69]: if you set a custom method the name should start `handle` prefix

File: components/upload/Upload.tsx [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+    const { onChange: onChangeProp } = props;
+    if (onChangeProp) {
+      onChangeProp({
+        ...info,
+        fileList: [...info.fileList],
+      });
+    }
+  };
+
+  const onStart = (file: RcFile) => {
```
Comment (igor-lemon) [inline components/upload/Upload.tsx:81]: if you set a custom method the name should start `handle` prefix

File: components/upload/Upload.tsx [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+      nextFileList[fileIndex] = targetItem;
+    }
+
+    onChange({
+      file: targetItem,
+      fileList: nextFileList,
+    });
+  };
+
+  const onSuccess = (response: any, file: UploadFile, xhr: any) => {
```
Comment (igor-lemon) [inline components/upload/Upload.tsx:100]: if you set a custom method the name should start `handle` prefix

File: components/upload/Upload.tsx [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    targetItem.status = 'done';
+    targetItem.response = response;
+    targetItem.xhr = xhr;
+    onChange({
+      file: { ...targetItem },
+      fileList: getFileList().concat(),
+    });
+  };
+
+  const onProgress = (e: { percent: number }, file: UploadFile) => {
```
Comment (igor-lemon) [inline components/upload/Upload.tsx:122]: if you set a custom method the name should start `handle` prefix

File: components/upload/Upload.tsx [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+
+        onChange({
+          file,
+          fileList: removedFileList,
+        });
+      }
+    });
+  };
+
+  const onFileDrop = (e: React.DragEvent<HTMLDivElement>) => {
```
Comment (igor-lemon) [inline components/upload/Upload.tsx:174]: if you set a custom method the name should start `handle` prefix

**Review by igor-lemon — CHANGES_REQUESTED** [review]

File: components/upload/Upload.tsx (hunk 2 above)
Comment (mouseluka) [reply components/upload/Upload.tsx:69]: discussed

File: components/upload/Upload.tsx (hunk 3 above)
Comment (mouseluka) [reply components/upload/Upload.tsx:81]: discussed

File: components/upload/Upload.tsx (hunk 4 above)
Comment (mouseluka) [reply components/upload/Upload.tsx:100]: discussed

File: components/upload/Upload.tsx (hunk 5 above)
Comment (mouseluka) [reply components/upload/Upload.tsx:122]: discussed

File: components/upload/Upload.tsx (hunk 6 above)
Comment (mouseluka) [reply components/upload/Upload.tsx:174]: discussed

**Review by igor-lemon — APPROVED** [review]

**Pull request merged**